src/state.rs

```diff
@@ -292,17 +292,35 @@ impl<EC: 'static + Debug, EA: 'static + Debug> AcmeState<EC, EA> {
     }
     async fn authorize(config: &AcmeConfig<EC, EA>, resolver: &ResolvesServerCertAcme, account: &Account, url: &String) -> Result<(), OrderError> {
         let auth = account.auth(&config.client_config, url).await?;
-        let (domain, challenge_url) = match auth.status {
+        let (domain, challenge_url, token) = match auth.status {
             AuthStatus::Pending => {
                 let Identifier::Dns(domain) = auth.identifier;
                 log::info!("trigger challenge for {}", &domain);
-                let (challenge, auth_key) = account.tls_alpn_01(&auth.challenges, domain.clone())?;
-                resolver.set_auth_key(domain.clone(), Arc::new(auth_key));
+                let challenge = match config.challenge_type {
+                    UseChallenge::Http01 => {
+                        let (challenge, key_auth) = account.http_01(&auth.challenges)?;
+                        resolver.set_key_auth(challenge.token.clone(), Arc::new(Vec::from(key_auth.as_ref())));
+                        challenge
+                    }
+                    UseChallenge::TlsAlpn01 => {
+                        let (challenge, auth_key) = account.tls_alpn_01(&auth.challenges, domain.clone())?;
+                        resolver.set_auth_key(domain.clone(), Arc::new(auth_key));
+                        challenge
+                    }
+                };
                 account.challenge(&config.client_config, &challenge.url).await?;
-                (domain, challenge.url.clone())
+                (domain, challenge.url.clone(), challenge.token.clone())
+            }
+            AuthStatus::Valid => {
```

Comment:
Can't we clear the whole auth key cache here instead? That could be a simple method. We should probably also clear the tls_alpn challenges here.

Reply:
can do that as well.